Adds Linear provider

jerriep opened the pull request:
Adds a Linear provider.

## Relevant docs

* [Linear OAuth 2.0 Authentication](https://developers.linear.app/docs/oauth/authentication)
* [Working with the GraphQL API](https://developers.linear.app/docs/graphql/working-with-the-graphql-api)

## Authorization flow screenshots

![image](https://github.com/user-attachments/assets/166e5a56-1824-4ea5-8311-9ebe331c0f83)

![image](https://github.com/user-attachments/assets/e0c0508d-c1a9-4497-8c62-68cff13f3a2f)

jerriep committed:
Adds Linear provider

jerriep commented on src/AspNet.Security.OAuth.Linear/LinearAuthenticationHandler.cs:
```diff
@@ -0,0 +1,102 @@
+        [NotNull] AuthenticationProperties properties,
+        [NotNull] OAuthTokenResponse tokens)
+    {
+        using var request = new HttpRequestMessage(HttpMethod.Post, "https://api.linear.app/graphql");
```
Linear does not have a user info endpoint and instead has a single GraphQL endpoint for **all** API requests. To retrieve the user info, you must build a GraphQL query and send it to the GraphQL endpoint.

Therefore, I did not consider it appropriate to use the usual pattern of setting and using the `Options.UserInformationEndpoint` property in the back channel. I am happy to hear suggestions on this.

martincostello commented on test/AspNet.Security.OAuth.Providers.Tests/Linear/bundle.json:
```diff
@@ -0,0 +1,37 @@
+      }
+    },
+    {
+      "comment": "https://docs.microsoft.com/en-us/linkedin/consumer/integrations/self-serve/sign-in-with-linkedin",
```
Needs updating.

martincostello commented on src/AspNet.Security.OAuth.Linear/LinearAuthenticationHandler.cs:
```diff
@@ -0,0 +1,102 @@
+    [JsonSerializable(typeof(GraphqlQuery))]
+    internal sealed partial class AppJsonSerializerContext : JsonSerializerContext;
+
+    internal sealed record GraphqlQuery([property: JsonPropertyName("query")] string Query);
```
Can these be `private`?

martincostello commented on src/AspNet.Security.OAuth.Linear/AspNet.Security.OAuth.Linear.csproj:
```diff
@@ -0,0 +1,18 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>$(DefaultNetCoreTargetFramework)</TargetFrameworks>
```
```suggestion
    <PackageValidationBaselineVersion>9.2.0</PackageValidationBaselineVersion>
    <!-- TODO Remove once published to NuGet.org -->
    <DisablePackageBaselineValidation>true</DisablePackageBaselineValidation>
    <TargetFrameworks>$(DefaultNetCoreTargetFramework)</TargetFrameworks>
```

martincostello reviewed (commented):
Could you add the new provider to the README table as well please?

jerriep committed:
Update src/AspNet.Security.OAuth.Linear/AspNet.Security.OAuth.Linear.csproj

Co-authored-by: Martin Costello <martin@martincostello.com>

jerriep committed:
Responds to code review

- Adds NuGet and developer docs to readme
- Updates incorrect comments in test bundle.json
- Makes source JSON serialization context and model private
